kernels/volk/volk_32f_reciprocal_32f.h

```diff
@@ -0,0 +1,228 @@
+
+    for (; number < eighth_points; number++) {
+        __m256 x = _mm256_load_ps(in);
+        in += 8;
+
+        __m256 x_inv = _mm256_rcp1_avx_ps(x);
+
+        _mm256_store_ps(out, x_inv);
+        out += 8;
+    }
+
+    number = eighth_points * 8;
+    for (; number < num_points; number++) {
+        *out++ = 1.f / (*in++);
+    }
+}
+#endif /* LV_HAVE_AVX */
+
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
+static inline void
+volk_32f_reciprocal_32f_a_avx_div(float* out, const float* in, unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int eighth_points = num_points / 8;
+    const __m256 ONE = _mm256_set1_ps(1.0f);
+
+    for (; number < eighth_points; number++) {
+        __m256 x = _mm256_load_ps(in);
+        in += 8;
+
+        __m256 x_inv = _mm256_div_ps(ONE, x);
+
+        _mm256_store_ps(out, x_inv);
+        out += 8;
+    }
+
+    number = eighth_points * 8;
+    for (; number < num_points; number++) {
+        *out++ = 1.f / (*in++);
+    }
+}
+#endif /* LV_HAVE_AVX */
+
+#endif /* INCLUDED_volk_32f_reciprocal_32f_a_H */
+
+#ifndef INCLUDED_volk_32f_reciprocal_32f_u_H
+#define INCLUDED_volk_32f_reciprocal_32f_u_H
+
+#if LV_HAVE_AVX2 && LV_HAVE_FMA
+#include <immintrin.h>
+#include <volk/volk_avx2_fma_intrinsics.h>
+static inline void
+volk_32f_reciprocal_32f_u_avx2_fma(float* out, const float* in, unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int eighth_points = num_points / 8;
+
+    for (; number < eighth_points; number++) {
+        __m256 x = _mm256_loadu_ps(in);
+        in += 8;
+
+        __m256 x_inv = _mm256_rcp1_avx2_fma_ps(x);
+
+        _mm256_storeu_ps(out, x_inv);
+        out += 8;
+    }
+
+    number = eighth_points * 8;
+    for (; number < num_points; number++) {
+        *out++ = 1.f / (*in++);
+    }
+}
+#endif /* LV_HAVE_AVX2 && LV_HAVE_FMA */
+
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
+#include <volk/volk_avx_intrinsics.h>
+static inline void
+volk_32f_reciprocal_32f_u_avx(float* out, const float* in, unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int eighth_points = num_points / 8;
+
+    for (; number < eighth_points; number++) {
+        __m256 x = _mm256_loadu_ps(in);
+        in += 8;
+
+        __m256 x_inv = _mm256_rcp1_avx_ps(x);
+
+        _mm256_storeu_ps(out, x_inv);
+        out += 8;
+    }
+
+    number = eighth_points * 8;
+    for (; number < num_points; number++) {
+        *out++ = 1.f / (*in++);
+    }
+}
+#endif /* LV_HAVE_AVX */
+
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
+static inline void
+volk_32f_reciprocal_32f_u_avx_div(float* out, const float* in, unsigned int num_points)
+{
+    unsigned int number = 0;
+    const unsigned int eighth_points = num_points / 8;
+    const __m256 ONE = _mm256_set1_ps(1.0f);
+
+    for (; number < eighth_points; number++) {
+        __m256 x = _mm256_loadu_ps(in);
+        in += 8;
+
+        __m256 x_inv = _mm256_div_ps(ONE, x);
```

Comment:
I wonder why you would just use a `div` in this kernel but implement an approximation for the other kernels.

Reply:
Ah, just for testing. Removed.